src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -249,16 +250,35 @@ private function formatListForPresentation(CombinationListForEditing $combinatio
                 'isSelected' => false,
                 'name' => $combination->getCombinationName(),
                 'reference' => $combination->getReference(),
-                //@todo: don't forget image path when implemented in the query
                 'impactOnPrice' => (string) $combination->getImpactOnPrice(),
                 'quantity' => $combination->getQuantity(),
                 'isDefault' => $combination->isDefault(),
+                'imageUrl' => $this->formatImageUrl($combination->getImagePath()),
             ];
         }
 
         return $data;
     }
 
+    /**
+     * @param string|null $imgPath
+     *
+     * @return string
+     */
+    private function formatImageUrl(?string $imgPath): string
+    {
+        $contextLink = $this->getContext()->link;
+
+        if (!$imgPath) {
+            $imageRetriever = new ImageRetriever($contextLink);
```

Comment:
I guess I cannot use this here? (it is adapter)
But then I would need to build the default image url manually (would need to use context link, context language code and build a string which looks smth like "en-default-small_default.jpg"). Not sure its any better either :sweat:

Reply:
This seems so complicated for something that could be hard coded in the twig 😅
But I see it's not that simple, it appears to be a generated image with localized text or something similar.. Maybe it's worth having a Twig extension to handle this

That could be used like `<img src="{{ noPitcureImageUrl('small') }}" />`